File: kata-setup-typescript/src/Item.Sulfuras.ts
import { Item, Itemer } from "./Item.js"

export class Sulfuras extends Item implements Itemer {
  constructor(sellIn: number, quality: number) {
    super("Sulfuras, Hand of Ragnaros", sellIn, quality)
  }

  updateQuality(newSellIn: number): void {
    return
  }

  updateSellIn(): void {
    return
  }
}

File: kata-setup-java/src/main/java/org/example/ItemSulfuras.java
package org.example;

public class ItemSulfuras extends Item {

  public ItemSulfuras(int sellIn, int quality) {
    super("Sulfuras, Hand of Ragnaros", sellIn, quality);
  }

  @Override
  public void decreaseSellIn() {
    // Nothing
  }

  @Override
  public void updateQuality() {
    // Nothing
  }
}
